dfastmi/io/map_file.py

```diff
@@ -0,0 +1,347 @@
+                ]
+                for mesh_attr in mesh_attrs:
+                    var_names = MapFile._get_var_names_from_var_attribute(mesh_variable, mesh_attr)
+                    for var_name in var_names:
+                        MapFile._copy_var(source_dataset, var_name, target_dataset)
+
+                        # check if variable has bounds attribute, if so copy those as well
+                        variable = source_dataset.variables[var_name]
+
+                        bounds_attr = "bounds"
+                        bounds_var_names = MapFile._get_var_names_from_var_attribute(variable, bounds_attr)
+                        for bounds_var_name in bounds_var_names:
+                            MapFile._copy_var(source_dataset, bounds_var_name, target_dataset)
+
+    @staticmethod
+    def _get_var_names_from_var_attribute(variable: nc.Variable, attr_name: str) -> List[str]:
+        try:
+            var_names = variable.getncattr(attr_name).split()
+        except AttributeError:
+            var_names = []
+            
+        return var_names
+        
+    def add_variable(
+        self,
+        variable_name: str,
+        data: np.array,
+        mesh_name: str,
+        face_dimension_name: str,
+        long_name: str,
+        unit: str,
+    ) -> None:
+        """
+        Add a new variable defined at faces to an existing UGRID netCDF file
+
+        Arguments
+        ---------
+        variable_name : str
+            Name of netCDF variable to be written.
+        data : numpy.array
+            Linear array containing the data to be written.
+        mesh_name : str
+            Name of mesh variable in the netCDF file.
+        face_dimension_name : str
+            Name of the face dimension of the selected mesh.
+        long_name : str
+            Long descriptive name for the variable ("None" if no long name attribute
+            should be written).
+        unit : str
+            String indicating the unit ("None" if no unit attribute should be written).
+        """
+        with nc.Dataset(self._map_file, "a") as dst:
+
+            var = dst.createVariable(variable_name, "f8", (face_dimension_name,))
+
+            var.mesh = mesh_name
+            var.location = "face"
+            var.long_name = long_name
+            var.units = unit
+            var[:] = data[:]
+
+        
+    def _get_node_coordinate_data(self, standard_names: List[str]) -> np.ndarray:
+        with nc.Dataset(self._map_file) as rootgrp:
+            mesh2d = rootgrp.variables[self.mesh2d_name]
+    
+            coord_var_names = mesh2d.getncattr("node_coordinates").split()
+            for coord_var_name in coord_var_names:
+                standard_name = rootgrp.variables[coord_var_name].standard_name
+                if standard_name in standard_names:
+                    var = rootgrp.variables[coord_var_name]
+                    break
+                
+            data = var[...]
```

Comment:
is this needed? would return var not work the same?

Reply:
var is a netcdf4 variable, with [...] the variable array is indexed to a numpy array.